R/zchunk_LB181.ag_R_C_Y_GLU_irr_mgmt.R

```diff
@@ -52,83 +54,351 @@ module_aglu_LB181.ag_R_C_Y_GLU_irr_mgmt_DISABLED <- function(command, ...) {
+    # Calculate yield index by iso/GLU/Irr_Rfd, as the harvest area-weighted yields to the global maximum 95th percentile yields
+    L181.Mueller_ag_Yield_tha_irr %>%
+      left_join_error_no_match(L181.Mueller_YieldMax_irr, by = c("GTAP_crop", "Irr_Rfd")) %>%
+      # Weighted by harvest area
+      mutate(wt_YieldIndex = hi * HA_ha / maxYield) %>%
+      group_by(iso, GLU, Irr_Rfd) %>%
+      summarise(HA_ha = sum(HA_ha), wt_YieldIndex = sum(wt_YieldIndex)) %>%
+      ungroup %>%
+      mutate(YieldIndex = wt_YieldIndex / HA_ha) ->
+      L181.YieldIndex_ctry_GLU_irr
+
+    # Next, figure out the 95th percentile observed yield across all regions and agricultural regions (intersection of country and GLU)
+    # Using 95th percentile because many of the crops have absurdly high maximum values (e.g., cucumbers with >500t/ha)
+    L181.noMueller_ag_Yield_tha_irr %>%
+      group_by(GTAP_crop) %>%
+      # Get the 95th percentile observed yield for each crop, across all regions and agricultural regions
+      summarise(maxYield_C = quantile(yield_tha, probs = 0.95)) ->
+      L181.noMueller_maxYield_crop
+
+    # Finally, set the "lo" and "hi" yields for crops not represented in the Mueller database
+    L181.noMueller_ag_Yield_tha_irr %>%
+      # Final "lo" yields are half of the observed
+      mutate(lo = yield_tha / 2) %>%
+      left_join_error_no_match(L181.noMueller_maxYield_crop, by = "GTAP_crop") %>%
+      # This creates NA, becuase not all countries, basins, and irrigation levels are necessarily represented in the yield index data
+      left_join(select(L181.YieldIndex_ctry_GLU_irr, -wt_YieldIndex, -HA_ha), by = c("iso", "GLU", "Irr_Rfd")) %>%
+      # Set NAs to zero
+      replace_na(list(YieldIndex = 0)) %>%
+      # Final "hi" yields are max between (95th percentile observed for each crop times iso/GLU/Irr_Rfd specific yield index)
+      # vs. (observed plus observed times min-yield-adj)
+      mutate(hi = pmax(maxYield_C * YieldIndex, yield_tha * (1 + MIN_YIELD_ADJ))) %>%
+      select(-maxYield_C, -YieldIndex) %>%
+      # Bind this back in with the dataset that only has Mueller crops
+      bind_rows(L181.Mueller_ag_Yield_tha_irr) ->
+      L181.ag_Yield_tha_ctry_crop_irr_mgmt
+
+    # In order to calculate weighted yield levels for aggregation, we don't want to be using the raw yields, as our
+    # GCAM commodities may include a blend of heterogeneous yielding commodities. For example, cucumber yields are in
+    # excess of 400 tonnes/hectare in some places, whereas pulses tend to be about 2. In a non-indexed aggregation,
+    # the cucumbers would be the only crop that matters for the final yields, and the yield of the "high" technology
+    # would not be representative of a biophysically attainable yield for the commodity class as a whole.
+
+    # Instead, two methods are developed to calculate the multipiers, and the data-system currently adopts Method TWO.
```

Comment:
Whoah. Hmm. We should get feedback from Page/Kate about whether this Method 1 should be retained or not. I'd really rather than not have the code here (and definitely not executing!) if it's not being used.